data-prepper-plugins/date-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/date/DateProcessor.java

```diff
@@ -146,11 +155,71 @@ private String getSourceTimestamp(final Record<Event> record) {
         }
     }
 
+    private Pair<String, Instant> getEpochFormatOutput(Instant time) {
+        if (outputFormat.equals("epoch_second")) {
+            return Pair.of(Long.toString(time.getEpochSecond()), time);
+        } else if (outputFormat.equals("epoch_milli")) {
+            return Pair.of(Long.toString(time.toEpochMilli()), time);
+        } else { // epoch_nano. validation for valid epoch_ should be
+                 // done at init time
+            long nano = (long)time.getEpochSecond() * 1000_000_000 + (long) time.getNano();
+            return Pair.of(Long.toString(nano), time);
+        } 
+    }
+
     private Pair<String, Instant> getFormattedDateTimeString(final String sourceTimestamp) {
+        ZoneId srcZoneId = dateProcessorConfig.getSourceZoneId();
+        ZoneId dstZoneId = dateProcessorConfig.getDestinationZoneId();
+        Long numberValue = null;
+        Instant epochTime;
+        
+        if (epochFormatters.size() > 0) {
+            try {
+                numberValue = Long.parseLong(sourceTimestamp);
+            } catch (NumberFormatException e) {
+                numberValue = null;
+            }
+        }
+        if (numberValue != null) {
+            int timestampLength = sourceTimestamp.length();
+            if (timestampLength > LENGTH_OF_EPOCH_IN_MILLIS) {
```

Comment:
I think we should invert these conditionals. Let's look at the list of `epoch_` values the user configures and then find the best fit from there.

This currently starts by inferring it and then seeing if the user has that. But, what if the value is very far in time from what we would typically expect?

Reply:
I think this is similar to providing two patterns like ["dd-mm-YYYY", "YYYY-mm-dd"] and the input not matching either one. Similarly, if the input has 13 digits and the pattern is ["epoch_second"] we should not match it. Basically, the pattern "epoch_second" should be treated as matching for 10-digit number, "epoch_milli" should be treated as matching for 13-digit number, and "epoch_nano" should be treated as matching 19-digit number. Because any other matching could result in silent errors.

Reply:
We discussed offline. We will be more lenient on going back in time, but not forward in time. And we will only allow the selection of only one of the `epoch_` configurations. Users won't select multiple of these for now.

Reply:
The current code already allows epoch times from the past but not from the future. Added check to make sure only one of the three "epoch_" patterns are allowed.